File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    
    const remaining = [];
    for (let i = 0; i < progresses.length; i++) {
        remaining.push(Math.ceil((100 - progresses[i]) / speeds[i]))
    }
    
    while (remaining.length > 0) {
        let count = 1;
        const popped = remaining.shift();
     
        while (remaining[0] <= popped) {
            remaining.shift();
            count++;
        }
        answer.push(count);
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math

def solution(progresses, speeds):
    answer = []
    days = [math.ceil((100 - progresses[i]) / speeds[i]) for i in range(len(progresses))]
    
    q = deque(days)
    
    while q:
        count = 1
        curr = q.popleft()
        while q and curr >= q[0]:
            q.popleft()
            count += 1
        answer.append(count)
        
    return answer